src/gr_sparse_mat/test/t-conversion.c

```diff
@@ -0,0 +1,168 @@
+/*
+    Copyright (C) 2024 Kartik Venkatram and Alden Walker
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <http://www.gnu.org/licenses/>.
+*/
+
+#include "test_helpers.h"
+#include "gr_sparse_mat.h"
+
+#define CHECK_TEST(x, name) { if (GR_SUCCESS != (x)) { flint_printf("FAIL %s\n", (name)); flint_abort(); } }
```

Comment:
Do not abort in the test function. Instead, let the main function for the test (i.e. `main.c`) take care of any aborts. So instead, just return some non-zero value to the main function and it will abort appropriately.

If you can, use `TEST_FUNCTION_FAIL`, but this has to be put into the top-level function in this file, i.e. under `TEST_FUNCTION_START(my_function, state)`.

Reply:
Also, this definition is in each file, which leads to a lot of code duplication.